src/hotspot/share/prims/perf.cpp

```diff
@@ -110,6 +110,7 @@ PERF_ENTRY(void, Perf_Detach(JNIEnv *env, jobject unused, jobject buffer))
 
   // get buffer address and capacity
   {
+   MACOS_AARCH64_ONLY(ThreadWXEnable wx(WXExec, thread));
```

Comment:
This looks like a workaround for another bug, or is it required here as well? Can this line be reverted?